dataplane/loader/src/main.rs

```diff
@@ -30,14 +32,15 @@ async fn main() -> Result<(), anyhow::Error> {
 
     info!("loading ebpf programs");
 
-    #[cfg(debug_assertions)]
-    let mut bpf_program = Ebpf::load(include_bytes_aligned!(
+    let path = if cfg!(debug_assertions) {
         "../../target/bpfel-unknown-none/debug/loader"
-    ))?;
-    #[cfg(not(debug_assertions))]
-    let mut bpf_program = Ebpf::load(include_bytes_aligned!(
+    } else {
         "../../target/bpfel-unknown-none/release/loader"
-    ))?;
+    };
```

Comment:
> The error was error cannot find value 'bpf-program' in this scope,because the variable bpf_program is only available when debug_assertions is enabled

Could you please tell me a bit more about how you originally triggered this problem?

> include_bytes_aligned! macro was unable to load the file at runtime. Replaced with fs::read() to load the file

So you're quite right that on a fresh clone of the repository the object file will be missing until the xtask is run to build the BPF object, and that's probably something we can improve in our tooling (and documentation). Again however, I'm curious about the original steps that triggered this: was the `Makefile` used or was it more like `cd dataplane; cargo build` that failed?

Reply:
I encountered a scope issue when running `make build.image.dataplane`, because the variable bpf_program is only available when debug_assertions is enabled resulting in an error. However, I noticed that the PR merged today (https://github.com/kubernetes-sigs/blixt/commit/427839a319483f958001ba47fa6b073a08ee7755) has fixed the issue.